v2/passwordless/nextjs/app-directory/protecting-route.mdx

```diff
@@ -173,23 +173,18 @@ export const TryRefreshComponent = () => {
 We then create a server component that can check if the session exists and return any session information we may need:
 
 ```tsx title="app/components/home.tsx"
-import { NextRequest, NextResponse } from "next/server";// typecheck-only, removed from output
-import { SessionContainer, VerifySessionOptions } from "supertokens-node/recipe/session";// typecheck-only, removed from output
-import { CollectingResponse } from "supertokens-node/framework/custom";// typecheck-only, removed from output
-declare let getSSRSession: (req?: NextRequest, options?: VerifySessionOptions) => Promise<{session: SessionContainer | undefined;hasToken: boolean;hasInvalidClaims: boolean;baseResponse: CollectingResponse;nextResponse?: NextResponse;}>; // typecheck-only, removed from output
-declare let TryRefreshComponent: any; // typecheck-only, removed from output
+import { getSSRSession } from "supertokens-node/nextjs";
+import { cookies, headers } from "next/headers";
 declare let SessionAuthForNextJS: any; // typecheck-only, removed from output
 import styles from "../page.module.css";
 import { redirect } from "next/navigation";
 // @ts-ignore
-import { getSSRSession } from "../sessionUtils";
-// @ts-ignore
 import { TryRefreshComponent } from "./tryRefreshClientComponent";
 // @ts-ignore
 import { SessionAuthForNextJS } from "./sessionAuthForNextJS";
 
 export async function HomePage() {
-    const { session, hasToken, hasInvalidClaims } = await getSSRSession();
+    const { session, hasToken, hasInvalidClaims } = await getSSRSession(cookies().getAll(), headers());
```

Comment:
need to surround it with try / catch in all places where we use getSSRSession, see https://github.com/supertokens/supertokens-node/pull/767#discussion_r1435008992

Reply:
done